prow/cmd/checkconfig/main.go

```diff
@@ -330,20 +330,7 @@ func validate(o options) error {
 		}
 	}
 	if o.warningEnabled(unknownFieldsWarning) {
-		cfgBytes, err := ioutil.ReadFile(o.configPath)
-		if err != nil {
-			return fmt.Errorf("error reading Prow config for validation: %w", err)
-		}
-		if err := validateUnknownFields(&config.Config{}, cfgBytes, o.configPath); err != nil {
-			errs = append(errs, err)
-		}
-	}
-	if pcfg != nil && o.warningEnabled(unknownFieldsWarning) {
-		pcfgBytes, err := ioutil.ReadFile(o.pluginConfig)
-		if err != nil {
-			return fmt.Errorf("error reading Prow plugin config for validation: %w", err)
-		}
-		if err := validateUnknownFields(&plugins.Configuration{}, pcfgBytes, o.pluginConfig); err != nil {
+		if _, err := config.LoadStrict(o.configPath, o.jobConfigPath); err != nil {
```

Comment:
The current validateUnknownFields was all fields except job fields, that seems like more of an oversight / artificial limitation / easier to implement rather than an intent to allow individually warning on unknown fields in only one of (plugins, prow, jobs).

just using `LoadStrict` here dedupes the logic if we ever decide to strict load in other components and covers all 3

I also have a diff stashed to enable it optionally in the config agent, but I think that's possibly another PR / probably this tool should switch to one-shot `Load` anyhow, but I want to keep the diff for this PR down.

Reply:
is it not actually a warning then?

Reply:
ah I see, there's only an option to convert all warnings to errors, there's no distinction between warning and error.
makes sense then.